topomodelx/utils/sparse.py

```diff
@@ -0,0 +1,32 @@
+"""Utils for more efficient sparse matrix casting to torch."""
+
+import numpy as np
+import torch
+from scipy.sparse import _csc
+
+
+def from_sparse(data: _csc.csc_matrix):
+    """Convert sparse input data directly to torch sparse coo format.
+
+    Parameters
+    ----------
+    data : scipy.sparse._csc.csc_matrix
+        Input n_dimensional data.
+
+    Returns
+    -------
+    torch.sparse_coo, same shape as data
+        input data converted to tensor.
+    """
+    if not isinstance(data, _csc.csc_matrix):
+        raise ValueError(
+            f"Expected Data type sparse._csc.csc_matrix, found {type(data)}"
+        )
```

Comment:
This check is not necessary. The type hint is sufficient and the code below would fail for other inputs anyway.

Reply:
And in case you want to keep the check, this should at least be a `TypeError`, not a `ValueError` :)

Reply:
Ah nice that type hint is enforced. Removed